src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -707,6 +708,16 @@ private function getWriteAccessInfo($class, $property, $value)
         $camelized = $this->camelize($property);
         $singulars = (array) Inflector::singularize($camelized);
 
+        if ($useAdderAndRemover && !\in_array($camelized, $singulars)) {
```

Comment:
unfortunately, this does not work for words where singular and plural are the same (aircraft for example)

Reply:
see the added test in #29355

Reply:
Yeah, there is a lot of exceptions [That are Both Plural and Singular](https://www.vappingo.com/word-blog/101-words-that-are-both-plural-and-singular/). Although I'm still confused why an empty array is passed to `getWriteAccessInfo()`. Just to implicitly allow adder/remover?

Reply:
Probably something like that, though I don't know for sure. Maybe the git history reveals some more background information.